google-cloud-pub-sub/src/main/scala/akka/stream/alpakka/googlecloud/pubsub/model.scala

```diff
@@ -147,48 +156,57 @@ object PublishMessage {
  * @param publishTime the time the message was published.
+ * @param orderingKey if non-empty, identifies related messages for which publish order should be respected
  */
 final class PubSubMessage private (val data: Option[String],
                                    val attributes: Option[immutable.Map[String, String]],
                                    val messageId: String,
-                                   val publishTime: Instant) {
+                                   val publishTime: Instant,
+                                   val orderingKey: Option[String]) {
 
   def withAttributes(attributes: java.util.Map[String, String]): PubSubMessage =
-    new PubSubMessage(data, Some(attributes.asScala.toMap), messageId, publishTime)
+    new PubSubMessage(data, Some(attributes.asScala.toMap), messageId, publishTime, orderingKey)
 
   def withData(data: String): PubSubMessage =
-    new PubSubMessage(Some(data), attributes, messageId, publishTime)
+    new PubSubMessage(Some(data), attributes, messageId, publishTime, orderingKey)
+
+  def withOrderingKey(orderingKey: String): PubSubMessage =
+    new PubSubMessage(data, attributes, messageId, publishTime, Some(orderingKey))
 
   override def equals(other: Any): Boolean = other match {
     case that: PubSubMessage =>
-      data == that.data && attributes == that.attributes && messageId == that.messageId && publishTime == that.publishTime
+      data == that.data && attributes == that.attributes && messageId == that.messageId && publishTime == that.publishTime && orderingKey == that.orderingKey
     case _ => false
   }
 
-  override def hashCode: Int = java.util.Objects.hash(data, attributes, messageId, publishTime)
+  override def hashCode: Int = java.util.Objects.hash(data, attributes, messageId, publishTime, orderingKey)
 
   override def toString: String =
-    "PubSubMessage(data=" + data + ",attributes=" + attributes + ",messageId=" + messageId + ",publishTime=" + publishTime + ")"
+    "PubSubMessage(data=" + data + ",attributes=" + attributes + ",messageId=" + messageId + ",publishTime=" + publishTime + ",orderingKey=" + orderingKey + ")"
 }
 
 object PubSubMessage {
 
   def apply(data: Option[String] = None,
             attributes: Option[immutable.Map[String, String]] = None,
             messageId: String,
-            publishTime: Instant) = new PubSubMessage(data, attributes, messageId, publishTime)
+            publishTime: Instant,
+            orderingKey: Option[String] = None) =
+    new PubSubMessage(data, attributes, messageId, publishTime, orderingKey)
 
   /**
    * Java API
    */
   def create(data: java.util.Optional[String],
              attributes: java.util.Optional[java.util.Map[String, String]],
              messageId: String,
-             publishTime: Instant) =
+             publishTime: Instant,
+             orderingKey: java.util.Optional[String]) =
     new PubSubMessage(Option(data.orElse(null)),
```

Comment:
To stay source compatible for Java users, you should add an overload for `create` with the new `orderingKey` and keep the existing `create` with its signature. (Just as you did with `PublishMessage`.)

Reply:
Done